tests/test_sqlalchemy.py

```diff
@@ -0,0 +1,279 @@
+
+    @testing.skip("databend")
+    def test_limit_offset_in_unions_from_alias(self):
+        pass
+
+
+class HasIndexTest(_HasIndexTest):
+    __requires__ = ('index_reflection',)
+
+
+class InsertBehaviorTest(_InsertBehaviorTest):
+    @testing.skip("databend")  # required autoinc columns
+    def test_insert_from_select_autoinc(self, connection):
+        pass
+
+    @testing.skip("databend")  # required autoinc columns
+    def test_insert_from_select_autoinc_no_rows(self, connection):
+        pass
+
+    @testing.skip("databend")  # required autoinc columns
+    def test_no_results_for_non_returning_insert(self, connection):
+        pass
+
+
+class LikeFunctionsTest(_LikeFunctionsTest):
+
+    @testing.skip("databend")
+    def test_contains_autoescape(self):
+        pass
+
+    @testing.skip("databend")
+    def test_contains_escape(self):
+        pass
+
+    @testing.skip("databend")
+    def test_contains_autoescape_escape(self):
+        pass
+
+    @testing.skip("databend")
+    def test_endswith_autoescape(self):
+        pass
+
+    @testing.skip("databend")
+    def test_endswith_escape(self):
+        pass
+
+    @testing.skip("databend")
+    def test_endswith_autoescape_escape(self):
+        pass
+
+    @testing.skip("databend")
+    def test_startswith_autoescape(self):
+        pass
+
+    @testing.skip("databend")
+    def test_startswith_escape(self):
+        pass
+
+    @testing.skip("databend")
+    def test_startswith_autoescape_escape(self):
+        pass
+
+
+class LongNameBlowoutTest(_LongNameBlowoutTest):
+    __requires__ = ("index_reflection",)  # This will do to make it skip for now
+
+
+class QuotedNameArgumentTest(_QuotedNameArgumentTest):
+    def quote_fixtures(fn):
+        return testing.combinations(
+            ("quote ' one",),
+            ('quote " two', testing.requires.symbol_names_w_double_quote),
+        )(fn)
+
+    @quote_fixtures
+    @testing.skip("databend")
+    def test_get_pk_constraint(self, name):
+        pass
+
+    @quote_fixtures
+    @testing.skip("databend")
+    def test_get_foreign_keys(self, name):
+        pass
+
+    @quote_fixtures
+    @testing.skip("databend")
+    def test_get_indexes(self, name):
+        pass
+
+
+class JoinTest(_JoinTest):
+    __requires__ = ("foreign_keys",)
+
+
+class BizarroCharacterFKResolutionTest(_BizarroCharacterFKResolutionTest):
+    __requires__ = ("foreign_keys",)
+
+
+class BinaryTest(_BinaryTest):
+
+    # ToDo - get this working, failing because cannot substitute bytes parameter into sql statement
+    # CREATE TABLE binary_test (x binary not null)
+    # INSERT INTO binary_test (x) values (b'7\xe7\x9f') ???
+    # It is possible to do this
+    # INSERT INTO binary_test (x) values (TO_BINARY('7\xe7\x9f'))
+    # but that's not really a solution I don't think
+    @testing.skip("databend")
+    def test_binary_roundtrip(self):
+        pass
+
+    @testing.skip("databend")
+    def test_pickle_roundtrip(self):
+        pass
```

Comment:
@hantmac Maybe you have a suggestion how I may get these binary tests working? Currently it is not possible to insert binary data using the dialect.

Reply:
@rad-pat Hi, the bitmap data type in Databend is a binary type that differs from other supported types in terms of its representation and display in SELECT statements. So you can use [this doc](https://docs.databend.com/sql/sql-reference/data-types/data-type-bitmap)